pkg/p2p/kadcast/packet_processor.go

```diff
@@ -0,0 +1,136 @@
+package kadcast
+
+import (
+	"encoding/binary"
+	"net"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/util/container/ring"
+)
+
+// ProcessTCPPacket recieves a Packet and processes it according to
+// it's type. It gets the packets from the circularqueue that
+// connects the listeners with the packet processor.
+func ProcessTCPPacket(queue *ring.Buffer, router *Router) {
+	// Instantiate now the variables to not pollute
+	// the stack.
+	var err error
+	var byteNum int
+	var senderAddr *net.UDPAddr
+	var tcpPayload []byte
+	var packet Packet
+	chunkIDmap := make(map[[16]byte]bool)
+	
+	for {
+		// Get all of the packets that are now on the queue.
+		queuePackets, _ := queue.GetAll()
+		for _, item := range queuePackets {
+			// Get items from the queue packet taken.
+			byteNum, senderAddr, tcpPayload, err = decodeRedPacket(item)
+			if err != nil {
+				log.WithError(err).Warn("Error decoding the TCP packet taken from the ring.")
+				continue
+			}
+			// Build packet struct
+			packet = getPacketFromStream(tcpPayload[:])
+			// Extract headers info.
+			tipus, senderID, nonce, peerRecepPort := packet.getHeadersInfo()
+
+			// Verify IDNonce
+			err = verifyIDNonce(senderID, nonce)
+			// If we get an error, we just skip the whole process since the
+			// Peer was not validated.
+			if err := verifyIDNonce(senderID, nonce); err != nil {
+				log.WithError(err).Warn("Incorrect TCP packet sender ID. Skipping its processing.")
+				continue
+			}
+
+			// Build Peer info and put the right port on it subsituting the one
+			// used to send the message by the one where the peer wants to receive
+			// the messages.
+			ip, _ := getPeerNetworkInfo(*senderAddr)
+			port := binary.LittleEndian.Uint16(peerRecepPort[:])
+			peerInf := MakePeer(ip, port)
+
+			switch tipus {
+			case 0: {
```

Comment:
No need for curly braces in a switch case, indentation is enough.

Reply:
Just thought it was more clear. 
Is it a bad practice??

Reply:
Not a bad practice, though I don't think it's particularly idiomatic, as I haven't seen it in any other Go code. Up to you :)